assets/js/modules/adsense/datastore/service.test.js

```diff
@@ -76,5 +94,51 @@ describe( 'module/adsense service store', () => {
 				expect( serviceURL ).toMatchQueryParameters( query );
 			} );
 		} );
+
+		describe( 'getServiceAccountSiteURL', () => {
+			beforeEach( () => {
+				registry.dispatch( CORE_SITE ).receiveSiteInfo( siteInfo );
+				registry.dispatch( STORE_NAME ).setSettings( settings );
+			} );
+
+			it( 'should return undefined if accountID is undefined', () => {
+				registry.dispatch( STORE_NAME ).setSettings( { accountID: undefined } );
+
+				const url = registry.select( STORE_NAME ).getServiceAccountSiteURL();
+				expect( url ).toBeUndefined();
+			} );
+
+			it( 'should return undefined if referenceSiteURL is undefined', () => {
+				registry.dispatch( CORE_SITE ).receiveSiteInfo( { referenceSiteURL: undefined } );
+
+				const url = registry.select( STORE_NAME ).getServiceAccountSiteURL();
+				expect( url ).toBeUndefined();
+			} );
+
+			it( 'should construct the correct `path` for the URL', () => {
+				const correctPath = `${ settings.accountID }/home`;
+
+				const resultingURL = registry.select( STORE_NAME ).getServiceAccountSiteURL();
+				const { pathname } = new URL( resultingURL );
+
+				expect( pathname.endsWith( correctPath ) ).toBe( true );
+			} );
+
+			it( 'should construct the correct query params for the URL', () => {
+				const { host: referenceSiteURL } = new URL( siteInfo.referenceSiteURL );
+
+				const resultingURL = registry.select( STORE_NAME ).getServiceAccountSiteURL();
+				expect( resultingURL ).toMatchQueryParameters( {
+					source: 'site-kit',
+					url: referenceSiteURL,
+				} );
+			} );
+
+			it( 'should append `urlParams` arguments to the `query` if received', () => {
+				const urlParams = { foo: 'bar' };
+				const url = registry.select( STORE_NAME ).getServiceAccountSiteURL( urlParams );
+				expect( url ).toMatchQueryParameters( urlParams );
+			} );
```

Comment:
This test doesn't show that the given `urlParams` are _appended_ to the default query params; only that they are present. Let's add another assertion to check that the default query params are also present.

It might also be good to add one more test here to show that `urlParams` take precedence over the default query params for the same key.